Domain/Services/ProductService.cs

```diff
@@ -88,7 +93,14 @@ public async Task<bool> UpdateProductAsync(ProductDTO dto)
                 throw new ValidationException(validationResult.Errors);
             }
 
-            return await _repository.Update(entity);
+            var result = await _repository.Update(entity);
+
+            if (result)
+            {
+                _ = _messagePublisher.PublishUpdateMessage(dto);
```

Comment:
looks like you can just return void from MessagePublisher instead of bool, since you are not using the result anywhere

Reply:
left as is to avoid waiting for it